src/Sylius/Bundle/CoreBundle/spec/EventListener/Workflow/Order/SaveAddressesOnCustomerListenerSpec.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\EventListener\Workflow\Order;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Customer\OrderAddressesSaverInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Symfony\Component\Workflow\Event\CompletedEvent;
+use Symfony\Component\Workflow\Marking;
+
+final class SaveAddressesOnCustomerListenerSpec extends ObjectBehavior
+{
+    function let(OrderAddressesSaverInterface $orderAddressesSaver): void
+    {
+        $this->beConstructedWith($orderAddressesSaver);
+    }
+
+    function it_throws_an_exception_on_non_supported_subject(\stdClass $callback): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('__invoke', [new CompletedEvent($callback->getWrappedObject(), new Marking())])
+        ;
+    }
+
+    function it_saves_addresses(
+        OrderAddressesSaverInterface $orderAddressesSaver,
+        OrderInterface $order,
+    ): void {
+        $event = new CompletedEvent($order->getWrappedObject(), new Marking());
+
+        $this($event);
+
+        $orderAddressesSaver->saveAddresses($order)->shouldHaveBeenCalledOnce();
```

Comment:
Just a general question, why is it sometimes `shouldBeCalled`, `shouldHaveBeenCalled` and `shouldHaveBeenCalledOnce`?

Reply:
i fixed that, only 2 specs should have this, the rest is just shouldBeCalled()